src/cockpit/channels/metrics.py

```diff
@@ -15,23 +15,192 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import asyncio
+import json
+import sys
+import time
 import logging
+from typing import Any, Dict, List, NamedTuple, Optional
 
-from ..channel import Channel
+from ..channel import AsyncChannel
+from .samples import Sampler, SAMPLE_DESCRIPTIONS
+from .samples import (cockpit_cpu_samples, cockpit_memory_samples, cockpit_cpu_temperature_samples)
 
 logger = logging.getLogger(__name__)
 
 
-class MetricsChannel(Channel):
+class Instance(NamedTuple):
```

Comment:
In cockpit we have normal metrics which are:

`memory.usage` => 8

But there are also metrics which are "multi-valued" (maybe a better word). Such as `cpu.nice` which has multiple values as: `{ 0: 1, 1: 2,... }` So that's an "instance". The protocol also calls them `instances`